exercises/yacht/Example.cs

```diff
@@ -0,0 +1,97 @@
+    ///    if entered in the Full House category.
+    ///
+    ///    Task
+    ///    Given a list of values for five dice and a category, your solution should return 
+    ///    the score of the dice for that category. If the dice do not satisfy the requirements
+    ///    of the category your solution should return 0. You can assume that five values will always be presented,
+    ///    and the value of each will be between one and six inclusively. You should not assume that the dice are ordered.
+    /// </summary>
+    /// <param name="diceSet">Dice Array</param>
+    /// <param name="category">Name of category</param>
+    /// <returns></returns>
+    public static int Score(int[] diceSet, string category)
+    {
+        switch (category.ToLower())
+        {
+            //Ones: Any combination The sum of dice with the number 1.
+            case "ones": return diceSet.Where(x => x == 1).Sum();
+            //Twos: Any combination The sum of dice with the number 2. 
+            case "twos": return diceSet.Where(x => x == 2).Sum();
+            //Threes: Any combination The sum of dice with the number 3.
+            case "threes": return diceSet.Where(x => x == 3).Sum();
+            //Fours: Any combination The sum of dice with the number 4.
+            case "fours": return diceSet.Where(x => x == 4).Sum();
+            //Fives: Any combination The sum of dice with the number 5.
+            case "fives": return diceSet.Where(x => x == 5).Sum();
+            //Sixes: Any combination The sum of dice with the number 6.
+            case "sixes": return diceSet.Where(x => x == 6).Sum();
+
+            //Full House:  Three of one number and two of another.  Sum of all dice.
+            case "full house":
+                {
+                    var testHashSet = new HashSet<int>();
+                    diceSet.ToList().ForEach(x => testHashSet.Add(x));
+                    return (testHashSet.Count() == 2) ? diceSet.Sum() : 0;
+                }
+            //Four-Of-A-Kind:  At least four dice showing the same face.   Sum of those four dice.
+            case "four-of-a-kind": // ?
+            case "four of a kind": // ?
+                var testDict = new Dictionary<int, int>();
+                diceSet.ToList().ForEach(x =>
+                {
+                    if (!testDict.ContainsKey(x)) testDict.Add(x, 1); else testDict[x]++;
+                });
+                if (testDict.Count > 2) return 0;
+                else if (testDict.Count == 1 || testDict.ElementAt(0).Value >= 4) return testDict.Keys.ToArray()[0] * 4;
+                else if (testDict.ElementAt(1).Value >= 4) return testDict.Keys.ToArray()[1] * 4;
+                else return 0;
+
+            //Little Straight: 1 - 2 - 3 - 4 - 5. Score varies. Often 30.
+            case "little straight": return (string.Join("-", diceSet.OrderBy(x => x)) == "1-2-3-4-5") ? 30 : 0;
```

Comment:
You believe could also write this slightly differently, without the need to convert to a string representation:

```csharp
diceSet.OrderBy(x => x).SequenceEqual(new[] { 1, 2, 3, 4, 5 })
```

Reply:
It doesn't compile: _Cannot implicitly convert type 'bool' to 'int'_

Reply:
I assume you replaced a little too much. The statement should become:

```csharp
return diceSet.OrderBy(x => x).SequenceEqual(new[] { 1, 2, 3, 4, 5 }) ? 30 : 0;
```